src/main/java/hotel/models/Booking.java

```diff
@@ -0,0 +1,92 @@
+package hotel.models;
+
+import customexceptions.InvalidHotelActionException;
+
+import java.awt.print.Book;
+import java.time.LocalDate;
+
+public class Booking {
+	private static int lastNumberUsed;
+	private final int bookingNumber;
+	private LocalDate fromDate;
+	private LocalDate toDate;
+	private String guestName;
+	private final long guestId;
+
+
+	/**
+	 * Constructor of the Booking class
+	 *
+	 * @param fromDate
+	 * @param toDate
+	 * @param guestName
+	 * @param guestId
+	 * @throws InvalidHotelActionException throws custom exception if any of the
+	 *                                     passed arguments is not valid for object creation
+	 */
+	public Booking(LocalDate fromDate, LocalDate toDate, String guestName, long guestId)
+		throws InvalidHotelActionException {
+		if (fromDate != null && toDate != null && guestName != null && guestId/1000000000L >= 1) {
```

Comment:
What is the idea behind this check?
`guestId/1000000000L >= 1`

Reply:
Checks if the number provided for the booking is 10 digits long - EGN

Reply:
If you have logic for EGN it might be good idea to extract it to a method where you can check it. Did you know that EGN can be CRC checked? Yes, it can - so you might not know the number but you can determine if it is correct or not.